plenum/test/node_catchup_with_3pc/test_catchup_with_skipped_commits.py

```diff
@@ -55,29 +42,53 @@ def check_lagging_node_is_not_syncing_audit():
     def check_lagging_node_done_catchup():
         assert lagging_node_state() == NodeLeecherService.State.Idle
 
+    def check_nodes_ordered_till(nodes: Iterable, view_no: int, pp_seq_no: int):
+        for node in nodes:
+            assert compare_3PC_keys((view_no, pp_seq_no), node.master_replica.last_ordered_3pc) >= 0
+
     # Preload nodes with some transactions
     sdk_send_random_and_check(looper, txnPoolNodeSet, sdk_pool_handle, sdk_wallet_client, 1)
     for node in txnPoolNodeSet:
         assert node.master_replica.last_ordered_3pc == (0, 1)
 
-    # Delay some commits on lagging node
-    some_commits = start_delaying(lagging_stasher, delay_some_commits(0, [2, 3]))
+    # Setup delayers
+    lagging_mid_commits = start_delaying(lagging_stasher, delay_3pc(after=3, before=6, msgs=Commit))
+    others_mid_commits = start_delaying(other_stashers, delay_3pc(after=3, before=6, msgs=Commit))
+    start_delaying(lagging_stasher, delay_3pc(before=4, msgs=Commit))
 
-    # Order 4 more requests in pool
-    sdk_send_random_and_check(looper, txnPoolNodeSet, sdk_pool_handle, sdk_wallet_client, 4)
+    # Send more requests
+    reqs = sdk_send_random_requests(looper, sdk_pool_handle, sdk_wallet_client, 6)
 
-    with delay_rules(lagging_stasher, delay_catchup(AUDIT_LEDGER_ID)):
-        # Start catchup
-        lagging_node.start_catchup()
-        looper.runFor(0.5)
-        assert lagging_node_state() == NodeLeecherService.State.SyncingAudit
+    # Wait until pool ordered till (0, 3)
+    looper.run(eventually(check_nodes_ordered_till, other_nodes, 0, 3))
+    assert lagging_node.master_replica.last_ordered_3pc == (0, 1)
 
-        # Allow some missing commits to be finally received
-        stop_delaying_and_process(some_commits)
-        looper.runFor(0.5)
+    with delay_rules(lagging_stasher, delay_catchup(DOMAIN_LEDGER_ID)):
+        with delay_rules(lagging_stasher, delay_catchup(AUDIT_LEDGER_ID)):
+            # Start catchup
+            lagging_node.start_catchup()
+            looper.runFor(0.5)
+            assert lagging_node_state() == NodeLeecherService.State.SyncingAudit
+
+            # Process missing commits on lagging node
+            stop_delaying_and_process(lagging_mid_commits)
+            looper.runFor(0.5)
+
+        # Allow to catchup audit ledger
+        looper.run(eventually(check_lagging_node_is_not_syncing_audit))
+        stop_delaying_and_process(others_mid_commits)
```

Comment:
Please make sure that the lagging node has a quorum of commits for (0,5) and (0,6) while it doesn't have it for (0, 4)

Reply:
Also I think we need to call `stop_delaying_and_process()` for `delay_3pc(before=4, msgs=Commit)`.
Then we need to wait for more than processOutOfOrderCommits timeout (1 sec?)